src/WebJobs.Extensions/Extensions/Timers/Scheduling/CronSchedule.cs

```diff
@@ -82,28 +86,72 @@ private DateTimeOffset GetNextOccurrence(DateTimeOffset now)
             // |      +1 hr     |                        |       -1 hr      |
             // | <invalid time> |                        | <ambiguous time> |
             //                  [========== DST ========>)
+            if (TryAdjustInvalidTime(next, out DateTimeOffset? adjusted))
+            {
+                // "Spring Forward": "next" was invalid and was adjusted forward to a valid time.
+                next = adjusted.Value;
+            }
+            else if (TryAdjustAmbiguousTime(now, next, out adjusted))
+            {
+                // "Fall Back": Either "now" or "next" was ambiguous and the time was adjusted.
+                next = adjusted.Value;
+            }
 
-            // Scenario 1 -- When Daylight Savings begins and the clock springs forward, we need to ensure that we are 
-            //   not scheduling a job to run at an invalid time. If the next occurrence is invalid, we need to skip
-            //   it and look for the next possible valid time.
-            bool isInvalidTime = timeZone.IsInvalidTime(nextDateTime);
+            return next;
+        }
+
+        /// <summary>
+        /// If the provided DateTime is not valid, moves forward to the next occurrence in the schedule
+        /// until a valid DateTime is found. For example, 02:30 does not exist when "springing forward"
+        /// to Daylight Savings Time in the Pacific Standard Time Zone. Uses TimeZoneInfo.Local to make
+        /// the determination.
+        /// </summary>
+        /// <param name="next">The DateTimeOffset to evaluate.</param>
+        /// <param name="adjusted">The adjusted DateTimeOffset, if "next" is invalid.</param>
+        /// <returns></returns>
+        internal bool TryAdjustInvalidTime(DateTimeOffset next, out DateTimeOffset? adjusted)
+        {
+            var nextDateTime = next.LocalDateTime;
+            bool isInvalidTime = TimeZoneInfo.Local.IsInvalidTime(nextDateTime);
+            adjusted = null;
+
+            if (!isInvalidTime)
+            {
+                return false;
+            }
 
             while (isInvalidTime)
             {
                 nextDateTime = _cronSchedule.GetNextOccurrence(nextDateTime);
-                isInvalidTime = timeZone.IsInvalidTime(DateTime.SpecifyKind(nextDateTime, DateTimeKind.Unspecified));
+                isInvalidTime = TimeZoneInfo.Local.IsInvalidTime(nextDateTime);
+            }
+
+            adjusted = new DateTimeOffset(nextDateTime);
+            return true;
+        }
+
+        internal bool TryAdjustAmbiguousTime(DateTimeOffset now, DateTimeOffset next, out DateTimeOffset? adjusted)
+        {
+            adjusted = null;
+
+            TimeZoneInfo timeZone = TimeZoneInfo.Local;
+            if (!timeZone.SupportsDaylightSavingTime)
```

Comment:
Can there be an invalid time if there is no DST? Do we need to add this check to ` TryAdjustInvalidTime` as well?Maybe it makes more sense to have this check be part of `GetNextOccurrence` before we do the "TryAdjust" bits

Reply:
I actually just refactored that as well -- this was just an optimization, but I moved it up to GetNextOccurrence now (as you suggested) so it's clearer that if there's no DaylightSavingTime, nothing else needs to happen.